bpf/init.sh

```diff
@@ -597,19 +606,17 @@ if [ "$MODE" == "ipvlan" ]; then
 	COPTS+=" -DENABLE_EXTRA_HOST_DEV"
 fi
 bpf_load $HOST_DEV1 "$COPTS" "egress" bpf_netdev.c bpf_host.o from-netdev $CALLS_MAP
-bpf_load $HOST_DEV1 "" "ingress" bpf_hostdev_ingress.c bpf_hostdev_ingress.o to-host $CALLS_MAP "no_qdisc_reset"
-
-IPSEC_LOAD_OPT=""
-if [ "$HOST_DEV1" == "$HOST_DEV2" ]; then
-	IPSEC_LOAD_OPT="no_qdisc_reset"
-fi
+bpf_load $HOST_DEV1 "" "ingress" bpf_hostdev_ingress.c bpf_hostdev_ingress.o to-host $CALLS_MAP
 # bpf_ipsec.o is also needed by proxy redirects, so we load it unconditionally
-bpf_load $HOST_DEV2 "" "ingress" bpf_ipsec.c bpf_ipsec.o from-netdev $CALLS_MAP $IPSEC_LOAD_OPT
+bpf_load $HOST_DEV2 "" "ingress" bpf_ipsec.c bpf_ipsec.o from-netdev $CALLS_MAP
```

Comment:
I *think* this breaks flannel: Flannel has `$HOSTDEV1` == `$HOST_DEV2`, and in that case you shouldn't replace the qdisc on the device, otherwise if I recall correctly you lose the *egress* program when replacing the qdisc.

Reply:
When doing qdisc replace, it keeps tc filters intact. Only adds the qdisc if it wasn't present.

Reply:
I withdraw my concerns then :-)